File: angular-frontend/src/app/models/parliamentMember-dto.ts
import { VotesDto } from './votes-dto';

export interface ParliamentMemberDto {
  active?: boolean;
  birthDate?: string;
  birthLocation?: string;
  club?: string;
  districtName?: string;
  districtNum?: number;
  educationLevel?: string;
  apiID?: number;
  email?: string;
  firstLastName?: string;
  firstName?: string;
  lastFirstName?: string;
  lastName?: string;
  numberOfVotes?: number;
  secondName?: string;
  voivodeship?: string;
  waiverDesc?: string;
  votes?: VotesDto[];
}

File: parliament-client/src/main/java/pl/sonmiike/parliamentclient/contract/ParliamentMemberDTO.java
package pl.sonmiike.parliamentclient.contract;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

import java.time.LocalDate;
import java.util.List;
import java.util.Optional;

@Data
@JsonInclude(JsonInclude.Include.NON_EMPTY)
public class ParliamentMemberDTO {

    private boolean active;
    private LocalDate birthDate;
    private String birthLocation;
    private String club;
    private String districtName;
    private String districtNum;
    private String educationLevel;
    private String email;
    private String firstLastName;
    private String firstName;
    @JsonProperty("id")
    private long apiId;
    private String lastFirstName;
    private String lastName;
    private String numberOfVotes;
    private String secondName;
    private String voivodeship;
    private Optional<String> waiverDesc = Optional.empty();
    private Optional<List<VotesDTO>> votes = Optional.empty();


}
